core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -224,15 +242,28 @@ public DirectoryWatcher(
    * is closed.
    */
   public void watch() {
+    final AtomicInteger eventCount = new AtomicInteger(0);
     while (listener.isWatching()) {
       // wait for key to be signalled
       WatchKey key;
       try {
-        key = watchService.take();
+        key = watchService.poll();
+        if (key == null) {
+          if (timeout > 0) {
+            currentOnIdleTimerTask = new OnIdleTimerTask(listener, eventCount.get());
```

Comment:
This creates a new OnIdleTimerTask each time. What is an OnIdleTimerTask already exists, does it need to be cancelled? Is all access got this field, and new instances to this field thread safe?